### PR TITLE
py-torchvision@0.5.0 depends on py-torch@1.4.1

### PR DESCRIPTION
**Pull request by rblake-llnl**
TLDR: We need python@2 with py-torch and py-torchvision.  The last working versions are py-torch@1.4.1 and py-torchvision@0.5.1.  This adds the missing version to py-torchvision.

See #20810 for why we can't build py-torch@1.4.0 with python@2

See #20801 for the original messages that caused this to be needed.

### CODE REVIEW

**Review by adamjstewart — CHANGES_REQUESTED** [review]
You also need to specify the `python` dependency version. Just change `0.5.0` to `0.5.0:0.5.999`

**Comment by adamjstewart** [comment]
Any idea how this works? I don't see a release or tag for v0.5.1 on GitHub.

**Comment by rblake-llnl** [comment]
Wow, I swore it used to be there.  I can still run: ```spack checksum py-torchvision 0.5.1```

I downloaded 0.5.1 manually from https://github.com/pytorch/vision/archive/v0.5.1.tar.gz and I can see tons of differences from https://github.com/pytorch/vision/archive/v0.5.0.tar.gz

```console
$ diff -r --brief vision-0.5.0/ vision-0.5.1/ | wc -l
157
```

**Comment by rblake-llnl** [comment]
Looked into it more.  Close and reject this branch, v0.5.1 has been called v0.6.0.

```
* b68adcf (HEAD, tag: v0.6.0, origin/v0.5.1) Make read_video_meta_data_from_memory and read_video_from_memory private (#2077) (#2084)
```

**Comment by rblake-llnl** [comment]
Adam, will this change instead let us use 1.4.1 with 0.5.0 ?

File: var/spack/repos/builtin/packages/py-torchvision/package.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class PyTorchvision(PythonPackage):
     # https://github.com/pytorch/vision#installation
     depends_on('python@3.6:', when='@0.7:', type=('build', 'run'))
     depends_on('python@3.5:', when='@0.6.0:0.6.999', type=('build', 'run'))
-    depends_on('python@2.7:2.8,3.5:3.8', when='@0.5.0', type=('build', 'run'))
+    depends_on('python@2.7:2.8,3.5:3.8', when='@0.5.0:0.5.999', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torchvision/package.py]: I would undo this change

File: var/spack/repos/builtin/packages/py-torchvision/package.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ class PyTorchvision(PythonPackage):
     depends_on('py-torch@1.6.0', when='@0.7.0', type=('build', 'link', 'run'))
     depends_on('py-torch@1.5.1', when='@0.6.1', type=('build', 'link', 'run'))
     depends_on('py-torch@1.5.0', when='@0.6.0', type=('build', 'link', 'run'))
-    depends_on('py-torch@1.4.0', when='@0.5.0', type=('build', 'link', 'run'))
+    depends_on('py-torch@1.4.0:1.4.999', when='@0.5.0', type=('build', 'link', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torchvision/package.py]: I would make it 1.4.1 instead, no reason to support 1.4.0 if it's broken. Technically these should all be ranges, but the developers don't seem to want to support that and I don't know what range will work.

**Review by adamjstewart — APPROVED** [review]
